dali/python/nvidia/dali/ops.py

```diff
@@ -367,12 +297,14 @@ def generate_outputs(self):
 
         if num_output == 0 and self._op.preserve:
             t_name = type(self._op).__name__ + "_id_" + str(self.id) + "_sink"
-            pipeline.add_sink(_EdgeReference(t_name, output_device, self))
+            pipeline.add_sink(_DataNode(t_name, output_device, self))
             return
 
         for i in range(num_output):
-            t_name = type(self._op).__name__ + "_id_" + str(self.id) + "_output_" + str(i)
-            t = _EdgeReference(t_name, output_device, self)
+            t_name = self._name
+            if num_output > 1:
+                t_name += "[{}]".format(i)
```

Comment:
Possible output name collision if the `_name` is duplicated - either requires more checking or additional mangling.